c7n/resources/backup.py

```diff
@@ -55,3 +56,47 @@ def get_resources(self, resource_ids, cache=True):
             except client.exceptions.ResourceNotFoundException:
                 continue
         return resources
+
+
+@resources.register('backup-vault')
+class BackupVault(QueryResourceManager):
+
+    class resource_type(TypeInfo):
+        service = 'backup'
+        enum_spec = ('list_backup_vaults', 'BackupVaultList', None)
+        detail_spec = ('describe_backup_vault', 'BackupVaultName', 'BackupVaultName', None)
+        name = id = 'BackupVaultName'
+        arn = 'BackupVaultArn'
+        universal_taggable = object()
+
+    def augment(self, resources):
+        super(BackupVault, self).augment(resources)
+        client = local_session(self.session_factory).client('backup')
+        results = []
+        for r in resources:
+            try:
+                tags = client.list_tags(ResourceArn=r['BackupVaultArn']).get('Tags', {})
```

Comment:
rather than doing an API Call per resource, we can replace this implementation/method with universal_augment which uses resource group tagging (fetch 20 resources tags per API Call).

Reply:
Replaced with universal_augment, updated test data, and signed the CLA. Let me know if there is anything else. Thanks!